data/shared/citizen/scripting/v8/main.js

```diff
@@ -47,6 +50,39 @@ const EXT_LOCALFUNCREF = 11;
 		binarraybuffer: true
 	});
 
+	// Vector2 unpacker
+	codec.addExtUnpacker(EXT_VECTOR2, (data) => {
+		// Converting Uint8Array to Buffer
+		const buffer = Buffer.from(data);
+		// Reading 3 float values from the buffer
+		const x = buffer.readFloatLE(0);
+		const y = buffer.readFloatLE(4);
+		return [x, y];
+	});
+
+	// Vector3 unpacker
+	codec.addExtUnpacker(EXT_VECTOR3, (data) => {
+		// Converting Uint8Array to Buffer
+		const buffer = Buffer.from(data);
+		// Reading 3 float values from the buffer
+		const x = buffer.readFloatLE(0);
+		const y = buffer.readFloatLE(4);
+		const z = buffer.readFloatLE(8);
+		return [x, y, z];
+	});
+
+	// Vector4 unpacker
+	codec.addExtUnpacker(EXT_VECTOR4, (data) => {
+		// Converting Uint8Array to Buffer
+		const buffer = Buffer.from(data);
+		// Reading 3 float values from the buffer
+		const x = buffer.readFloatLE(0);
+		const y = buffer.readFloatLE(4);
+		const z = buffer.readFloatLE(8);
+		const w = buffer.readFloatLE(12);
+		return [x, y, z, w];
+	});
+
```

Comment:
This is equivalent.

```js
const vectorUnpacker = (data => [...new Float32Array(data.buffer)]);

codec.addExtUnpacker(EXT_VECTOR2, vectorUnpacker);
codec.addExtUnpacker(EXT_VECTOR3, vectorUnpacker);
codec.addExtUnpacker(EXT_VECTOR4, vectorUnpacker);
```

And this one _should_ improve number precision.

```js
const vectorUnpacker = (data => Array.from(new Float32Array(data.buffer), (v) => Number(v.toPrecision(7))));
```

Reply:
Thanks, @thelindat! That’s a great idea, and I really like the improvement for precision. I’ll go ahead and push this change.